hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/burn/BurnCall.java

```diff
@@ -16,6 +16,69 @@
 
 package com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.burn;
 
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.revertResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCall.PricedResult.gasOnly;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.asHeadlongAddress;
+
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.contract.impl.exec.gas.DispatchType;
+import com.hedera.node.app.service.contract.impl.exec.gas.SystemContractGasCalculator;
+import com.hedera.node.app.service.contract.impl.exec.scope.SystemContractOperations;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategy;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AddressIdConverter;
 import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCall;
+import com.hedera.node.app.service.token.records.TokenBurnRecordBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.function.LongFunction;
+import java.util.function.Supplier;
+
+public interface BurnCall extends HtsCall {
+    /**
+     * Performs the {@link HtsCall#execute()} logic shared between fungible and non-fungible burns.
+     *
+     * @param tokenId the token ID, if found
+     * @param bodySupplier the function to create the transaction body
+     * @param dispatchType the dispatch type
+     * @param addressIdConverter the address ID converter
+     * @param verificationStrategy the verification strategy
+     * @param gasCalculator the gas calculator
+     * @param spender the spender
+     * @param systemContractOperations the system contract operations
+     * @param invalidTokenReversion the invalid token reversion
+     * @return the result of the call
+     */
 
-public interface BurnCall extends HtsCall {}
+    // too may parameters
```

Comment:
"may" -> "many" but thanks for the comment! otherwise these sonar suppressions are completely opaque and you can't even tell when they're no longer necessary (when the code gets rewritten, they just tend to hang around)

(Plus this is one of the mildly stupid ones, and it should be configurable as to N to adapt to local coding styles for specific projects.)

Reply:
Fixed. 👍

Reply:
(not actually fixed - but IDC)